### PR TITLE
feat: introduce support for Talos API access from Kubernetes

### PR DESCRIPTION
**Pull request by smira**
This is a first step: providing a service to access Talos API.

See #4422 

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/siderolabs/talos/5830)

### CODE REVIEW

File: internal/app/machined/pkg/controllers/kubeaccess/endpoint.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+		} else {
+			newEndpoints = oldEndpoints.DeepCopy()
+		}
+
+		newEndpoints.Subsets = []corev1.EndpointSubset{
+			{
+				Ports: []corev1.EndpointPort{
+					{
+						Name:     "apid",
+						Port:     50000,
```
Comment (utkuozdemir) [inline internal/app/machined/pkg/controllers/kubeaccess/endpoint.go:212]: Maybe use `constants.ApidPort` instead?

**Review by utkuozdemir — APPROVED** [review]
LGTM

File: internal/app/machined/pkg/controllers/kubeaccess/endpoint.go (hunk 1 above)
Comment (smira) [reply internal/app/machined/pkg/controllers/kubeaccess/endpoint.go]: fixed!

File: internal/app/machined/pkg/controllers/kubeaccess/endpoint.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+		var newEndpoints *corev1.Endpoints
+
+		if apierrors.IsNotFound(err) {
+			newEndpoints = &corev1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      constants.KubernetesTalosAPIServiceName,
+					Namespace: constants.KubernetesTalosAPIServiceNamespace,
+					Labels: map[string]string{
+						"provider":  "talos",
```
[the remark this answers is not in the document]
Comment (smira) [reply internal/app/machined/pkg/controllers/kubeaccess/endpoint.go]: fixed!

**Review by sauterp — APPROVED** [review]

**Comment by smira** [comment]
I did some cosmetic fixes - added config validation unit-tests, changed `bool` to be `*bool`, added a config example, etc.

**Review by frezbo — APPROVED** [review]
any release notes needed, or only when the whole thing is completed?

**Comment by smira** [comment]
> any release notes needed, or only when the whole thing is completed?

hopefully when the full thing is done - we'll need also some docs

**Pull request merged**